pgmpy/factors/Factor.py

```diff
@@ -127,6 +127,9 @@ def get_cardinality(self, variables):
         >>> phi.get_cardinality(['x1', 'x2'])
         {'x1': 2, 'x2': 3}
         """
+        if isinstance(variables, str):
+            raise TypeError("List or array-like parameter expected, given parameter {variables} is type str".format(variables=variables))
```

Comment:
@yashu-seth This seems to be a better more clearer message to me: `variables: Expected type list or array-like, got type str`. And don't put the actual parameter value in the message, will just be confusing. What do you think?